clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -1291,9 +1294,21 @@ class CastPlacementVisitor :
                    IsChecked Dinfo) {
     auto &E = Info.getConstraints().getVariables();
     auto SrcChecked = Src->anyChanges(E);
-    // Check if the src is a checked type and destination is not.
-    return (SrcChecked && !Dst->anyChanges(E)) ||
-           (SrcChecked && Dinfo == WILD);
+    // Check if the src is a checked type.
+    if (SrcChecked) {
+      // Check if Dst is an itype, if yes then
+      // Src should have exactly same checked type else we need to insert cast.
+      if (Dst->getItypePresent()) {
+        return !Dst->haveSameAssignment(Info.getConstraints(), Src);
+      }
+
```

Comment:
This is basically the stuff you added for the FP conversion, right? The `surroundByCast` and `FancyCast` stuff is just a reworking of what you had before?

Reply:
Yes.